src/components/Detail/Detail.jsx

```diff
@@ -0,0 +1,128 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate, useParams } from 'react-router-dom';
+import { fetchYachtDetails } from '../../redux/detailsSlice';
+import { Link } from 'react-router-dom';
+import { FaAngleRight, FaSun, FaArrowAltCircleRight } from 'react-icons/fa';
+import multicolor from '../../assets/multicolor.png';
+import './Detail.css';
+import Loading from '../Loading/Loading';
+
+const Details = () => {
+  const navigate = useNavigate();
+  const trStyle = {
+    textAlign: 'center',
+    fontWeight: '400',
+    fontStyle: 'italic',
+  };
+  const navigateToReservation = () => {
+    navigate('/reserve', { state: yachtDetails });
+  };
+
+  const { id } = useParams();
+  const dispatch = useDispatch();
+  const yachtDetails = useSelector((state) => state.details.yachtDetails);
+  const loading = useSelector((state) => state.details.isLoading);
+  const hasErrors = useSelector((state) => state.details.hasErrors);
+
+  useEffect(() => {
+    dispatch(fetchYachtDetails(id));
+  }, [dispatch, id]);
+
+  if (hasErrors) {
+    return <div>There was an error fetching yacht details.</div>; // Correct error message
+  }
+
+  function annualPercentage() {
```

Comment:
- [ ] Kindly, refactor this function to ES6.   The "function" keyword in the industry is soft deprecated almost everywhere. 👍